### PR TITLE
[TEAM2-520] Allow Crosschain Token Selection in Get $ASSET Expanded State Flow

### PR DESCRIPTION
**Pull request by derHowie**
## What changed (plus any additional context for devs)
The 'Get ${asset}' flow from expanded state did not previously support crosschain selections and would clear the output in the ExchangeModal. This now works.

## Screen recordings / screenshots
Bridge Case: https://recordit.co/MWdPKfXc9n
Crosschain: https://recordit.co/yh6i0iZXxS
Single chain: https://recordit.co/Fd9VGmxVHt

## What to test
Ensure the cases above behave appropriately and swaps are possible in each case after proceeding through the 'Get ${asset}' flow.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/screens/ExchangeModal.tsx [hunk 1]
```diff
@@ -303,6 +266,44 @@ export default function ExchangeModal({
...
+  // we want to update the output to be on the same, if its not available -> null
+  const defaultOutputAssetOverride = useMemo(() => {
+    const newOutput = defaultOutputAsset;
+
+    if (
+      defaultInputAsset &&
+      defaultOutputAsset &&
+      defaultInputAsset.type !== defaultOutputAsset.type
+    ) {
+      const crosschainImplementation =
```
Comment (estebanmino) [inline src/screens/ExchangeModal.tsx:279]: very minor, feels like `crosschainImplementationAddress` is more accurate?

Comment (estebanmino) [reply src/screens/ExchangeModal.tsx:279]: also minor, https://github.com/rainbow-me/rainbow/pull/4281/files#diff-bcc39e0a9afb8c2146a721fb287948c2c1835246c62b2509304194f144308082L234 seems like is the same code than before crosschain i got confused that this was only for crosschain swaps

**Review by estebanmino — APPROVED** [review]
code lgtm, got confused about variable names but is working for all cases and also tested with crosschain flag off to maintain the behavior 👌

File: src/screens/ExchangeModal.tsx [hunk 2]
```diff
@@ -303,6 +266,46 @@ export default function ExchangeModal({
...
+  const defaultOutputAssetOverride = useMemo(() => {
+    const newOutput = defaultOutputAsset;
+
+    if (
+      defaultInputAsset &&
+      defaultOutputAsset &&
+      defaultInputAsset.type !== defaultOutputAsset.type
+    ) {
+      // find address for output asset on the input's network
+      // TODO: this value can be removed after the crosschain swaps flag is no longer necessary
```
Comment (skylarbarrera) [inline src/screens/ExchangeModal.tsx:280]: 🤌

**Review by skylarbarrera — APPROVED** [review]
🦧

**Pull request merged**